### PR TITLE
Feature/application support

### PR DESCRIPTION
**Pull request by danhanson**
If running in the in-memory-server, you should add code into io.pathfinder.config.Global in order to have an application to test with(don't commit it though):

```
    public void onStart(Application app) {
        Logger.info("Application has started.");
        PathFinderApplication pfApp = new PathFinderApplication();
        pfApp.name_$eq("TEST APP");
        pfApp.insert();
        Cluster c = new Cluster();
        c.pathFinderApplication_$eq(pfApp);
        c.insert();
        pfApp.defaultCluster_$eq(c);
        c.save();
        pfApp.save();
        Logger.info("Created TEST APP with key: "+pfApp.id());
        Router.init();
    }
```

Otherwise, you can add the application through the pathfinder website.
you can then use the application id to make a request:

```
webSocket.send(JSON.stringify(
  { "getClusters": {"id": "001e7047-ee14-40d6-898a-5acf3a1cfd8a"} }
));
```

Which responds with the following message of the form:

```
{ "clusters": {
  "id": "001e7047-ee14-40d6-898a-5acf3a1cfd8a",
  "defaultClusterId": 10,
  "clusterIds": [10, 20, 30, 45]
}}
```

### CODE REVIEW

**Comment by danhanson** [comment]
It appears that adding the application model and relationship between clusters and applications broke a bunch of seemingly unrelated tests because Ebean's dumb

**Comment by aj-michael** [comment]
Wait what? I have to add additional code after cloning the repo to make it run in dev mode? Why?

**Comment by aj-michael** [comment]
Oh I see what you're saying. You're saying that there is no other way to get an application in the DB.

I think there's a better way to do this. I think we should set it up so that if you start the server with:

```
play run -Dconfig.resource=dev.conf
```

it loads a conf file that initializes the db.

File: app/io/pathfinder/models/PathFinderApplication.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    @Id
+    @Column(updatable = false)
+    var id: UUID = null
+
+    @Column
+    var name: String = null
+
+    @OneToOne
+    @JoinColumn
+    var defaultCluster: Cluster = null
```
Comment (robinsdm) [inline app/io/pathfinder/models/PathFinderApplication.scala]: rename field to defaultClusterId

**Comment by aj-michael** [comment]
cool

**Pull request merged**